src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -68,107 +68,199 @@ def validate_location(namespace):
+        return 'specialized managed disk'
+
+
+storage_profile_param_options = {
+    'os_disk_name': '--os-disk-name',
+    'storage_caching': '--storage-caching',
+    'os_type': '--os-type',
+    'attach_os_disk': '--attach-os-disk',
+    'image': '--image',
+    'storage_account': '--storage-account',
+    'storage_container_name': '--storage-container-name',
+    'storage_sku': '--storage-sku',
+    'use_unmanaged_disk': '--use-unmanaged-disk'
+}
+
+
+def _validate_required_forbidden_parameters(namespace, required, forbidden):
+    missing_required = [x for x in required if not getattr(namespace, x)]
+    included_forbidden = [x for x in forbidden if getattr(namespace, x)]
+    if missing_required or included_forbidden:
+        error = 'invalid usage for storage profile: {}:'.format(
+            _get_storage_profile_description(namespace.storage_profile))
+        if missing_required:
+            missing_string = ', '.join(
+                storage_profile_param_options[x] for x in missing_required)
+            error = '{}\n\tmissing: {}'.format(error, missing_string)
+        if included_forbidden:
+            forbidden_string = ', '.join(
+                storage_profile_param_options[x] for x in included_forbidden)
+            error = '{}\n\tnot applicable: {}'.format(error, forbidden_string)
+        raise CLIError(error)
+
+
+def _validate_managed_disk_sku(sku):
+
+    allowed_skus = ['Premium_LRS', 'Standard_LRS']
+    if sku.lower() not in [x.lower() for x in allowed_skus]:
+        raise CLIError("invalid storage SKU '{}': allowed values: '{}'".format(sku, allowed_skus))
+
+
+# pylint: disable=too-many-branches, too-many-statements, redefined-variable-type
+def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
+
+    # use minimal parameters to resolve the expected storage profile
+    if getattr(namespace, 'attach_os_disk', None) and not namespace.image:
+        if namespace.use_unmanaged_disk:
+            # STORAGE PROFILE #3
+            namespace.storage_profile = StorageProfile.SASpecializedOSDisk
         else:
-            # last try: is it a custom image name?
-            compute_client = _compute_client_factory()
-            try:
-                compute_client.images.get(namespace.resource_group_name, image)
-                image = namespace.image = _get_resource_id(image, namespace.resource_group_name,
-                                                           'images', 'Microsoft.Compute')
-                namespace.storage_profile = StorageProfile.ManagedCustomImage
-            except CloudError:
-                err = 'Invalid image "{}". Use a custom image name, id, or pick one from {}'
-                raise CLIError(err.format(image, [x['urnAlias'] for x in images]))
+            # STORAGE PROFILE #6
+            namespace.storage_profile = StorageProfile.ManagedSpecializedOSDisk
+    elif namespace.image and not getattr(namespace, 'attach_os_disk', None):
```

Comment:
It is, but I do the validation in two passes. 1) an efficient "guess" at the profile type based on key parameters and then 2) a specific validation for that storage profile. This check will give a more specific error than the "_validate_required_forbidden_parameters" check.